addons/point_of_sale/models/pos_session.py

```diff
@@ -124,8 +127,10 @@ def _compute_cash_balance(self):
 
     @api.depends('order_ids.payment_ids.amount')
     def _compute_total_payments_amount(self):
+        result = self.env['pos.payment'].read_group([('session_id', 'in', self.ids)], ['amount'], ['session_id'])
+        session2amount = dict((data['session_id'][0], data['amount']) for data in result)
```

Comment:
it looks good for me, maybe just don't name variable with numbers in it.

Reply:
Many2one, Many2many, One2many don't follow this rule :wink:

Reply:
r+ ?

Reply:
r+ ? :grimacing:

Reply:
it looks good for me, maybe just don't name variable with numbers in it.

Reply:
Done

Reply:
r+ ?